src/main/java/blusunrize/immersiveengineering/client/utils/ClocheRenderHelper.java

```diff
@@ -0,0 +1,217 @@
+		IntegerProperty ageProperty;
+
+		public RenderFunctionCrop(Block cropBlock)
+		{
+			this.cropBlock = cropBlock;
+			if(cropBlock instanceof CropsBlock)
+			{
+				this.maxAge = ((CropsBlock)cropBlock).getMaxAge();
+				this.ageProperty = ((CropsBlock)cropBlock).getAgeProperty();
+			}
+			else
+			{
+				for(IProperty<?> prop : cropBlock.getDefaultState().getProperties())
+					if("age".equals(prop.getName())&&prop instanceof IntegerProperty)
+					{
+						int tmp = -1;
+						for(Integer allowed : ((IntegerProperty)prop).getAllowedValues())
+							if(allowed!=null&&allowed > tmp)
+								tmp = allowed;
+						if(tmp > 0)
+						{
+							this.maxAge = tmp;
+							this.ageProperty = (IntegerProperty)prop;
+							break;
+						}
+					}
+			}
+			if(this.ageProperty==null||this.maxAge <= 0)
+				throw new IllegalArgumentException("Block "+cropBlock.getTranslationKey()+" is not a valid crop block");
+		}
+
+		@Override
+		public float getScale(ItemStack seed, float growth)
+		{
+			return 0.875f;
+		}
+
+		@Override
+		public Collection<Pair<BlockState, TRSRTransformation>> getBlocks(ItemStack stack, float growth)
+		{
+			int age = Math.min(this.maxAge, Math.round(this.maxAge*growth));
+			return ImmutableList.of(Pair.of(this.cropBlock.getDefaultState().with(this.ageProperty, age), DEFAULT_TRANSFORMATION));
+		}
+	}
+
+	public static class RenderFunctionStacking implements ClocheRenderFunction
+	{
+		final Block cropBlock;
+
+		public RenderFunctionStacking(Block cropBlock)
+		{
+			this.cropBlock = cropBlock;
+		}
+
+		@Override
+		public float getScale(ItemStack seed, float growth)
+		{
+			return 0.6875f;
+		}
+
+		@Override
+		public Collection<Pair<BlockState, TRSRTransformation>> getBlocks(ItemStack stack, float growth)
+		{
+			TRSRTransformation bottom = new TRSRTransformation(new Vector3f(0, growth-1, 0), null, null, null);
+			TRSRTransformation top = new TRSRTransformation(new Vector3f(0, growth, 0), null, null, null);
+			return ImmutableList.of(
+					Pair.of(this.cropBlock.getDefaultState(), bottom),
+					Pair.of(this.cropBlock.getDefaultState(), top));
+		}
+	}
+
+	public static class RenderFunctionStem implements ClocheRenderFunction
+	{
+		final StemGrownBlock cropBlock;
+		final StemBlock stemBlock;
+		final AttachedStemBlock attachedStemBlock;
+
+		public RenderFunctionStem(Block cropBlock)
+		{
+			Preconditions.checkArgument(cropBlock instanceof StemGrownBlock);
+			this.cropBlock = (StemGrownBlock)cropBlock;
+			this.stemBlock = this.cropBlock.getStem();
+			this.attachedStemBlock = this.cropBlock.getAttachedStem();
+		}
+
+		@Override
+		public float getScale(ItemStack seed, float growth)
+		{
+			return 1;
+		}
+
+		@Override
+		public Collection<Pair<BlockState, TRSRTransformation>> getBlocks(ItemStack stack, float growth)
+		{
+			Set<Pair<BlockState, TRSRTransformation>> ret = new HashSet();
+			Matrix4f stemMatrix = new Matrix4f();
+			stemMatrix.setIdentity();
+			stemMatrix.rotY(-1.571f); //-90 deg
```

Comment:
Something like `-(float)Math.PI/2` would be clearer IMO.

Reply:
I didn't want to include unncecessary maths, but I suppose Java optimises this to a constant anyway, huh?

Reply:
I'm quite sure it actually optimizes it at compile-time (not JIT), ForgeFlower (Forge's FernFlower fork) "recently" had to add something to replace that sort of thing with something readable.